### PR TITLE
Update fastqc to adhere to new module guidelines

### PR DESCRIPTION
**Pull request by grst**
- [x] Module file should only define inputs/outputs as parameters and have the ability to use params.MODULENAME_options as an additional parameter to add any additional settings via pipelines.
   - ~~Specify single-end boolean values within the input channel and not be inferred from the data e.g. here~~
(fastqc does not care about single/paired end)
   - [x] Define threads or resources where required for a particular process using task.cpus
   - [x] Software that can be piped together should be added to separate module files unless there is an run-time, storage advantage in implementing in this way e.g. bwa mem | samtools view to output BAM instead of SAM - Process names should be all uppercase
 - [x] The publishDirMode should be configurable via params.publish_dir_mode
- [x] Test data is stored within this repo. Re-use generic files from tests/data by symlinking them into the test directory of the module. Add specific files to the test-directory directly. Keep test files as tiny as possible.
 - [x] Software requirements should be declared in a conda environment.yml file, including exact version numbers. Additionally, there should be a Dockerfile that containerizes the environment.
- [x] Each process should emit a file TOOL.version.txt containing a single line with the software's version in the format vX.X.X.
    All outputs should be named

### CODE REVIEW

**Comment by ewels** [comment]
Scooped up your other PR here somehow I think..

**Comment by grst** [comment]
ahhrg... thanks, I'll fix it!

File: software/fastqc/meta.yml [hunk 1]
```diff
@@ -1,33 +1,48 @@
...
+params:
+  - fastqc_args:
+      type: string
+      description: Additional command line arguments passed to fastqc.
+  - out_dir:
+      type: string
+      description: |
+        The pipeline's output directory. By default, the module will
+        output filese into `$out_dir/MODULE_NAME`
+  - fastqc_publish_dir:
```
Comment (drpatelh) [inline software/fastqc/meta.yml:26]:
```suggestion
  - publish_dir:
```

File: software/fastqc/meta.yml [hunk 2]
```diff
@@ -1,33 +1,48 @@
...
+      documentation: https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/
+params:
+  - fastqc_args:
+      type: string
+      description: Additional command line arguments passed to fastqc.
+  - out_dir:
+      type: string
+      description: |
+        The pipeline's output directory. By default, the module will
+        output filese into `$out_dir/MODULE_NAME`
```
Comment (drpatelh) [inline software/fastqc/meta.yml:25]:
```suggestion
        output files into `$out_dir/MODULE_NAME`
```

File: software/fastqc/meta.yml [hunk 3]
```diff
@@ -1,33 +1,48 @@
...
+  - fastqc_publish_dir:
+      type: string
+      description: |
+        Override the automatically generated output directory.
```
Comment (drpatelh) [inline software/fastqc/meta.yml:29]:
```suggestion
        Append to the path for the standard output directory provided by `$out_dir`.
```

File: software/fastqc/meta.yml [hunk 4]
```diff
@@ -1,33 +1,48 @@
...
+  - publish_dir_mode:
+      type: string
+      description: Set the publishDir mode (e.g. copy, link, ...)
```
Comment (drpatelh) [inline software/fastqc/meta.yml:32]:
```suggestion
      description: Provide a value for the Nextflow `publishDir` mode parameter (e.g. copy, link, ...)
```

File: software/fastqc/meta.yml [hunk 5]
```diff
@@ -1,33 +1,48 @@
...
-    -
-        - name:
-            type: string
-            description: Sample identifier
-        - reads:
-            type: file
-            description: Input FastQ file, or pair of files
+  - name:
+      type: string
+      description: Sample identifier
```
Comment (drpatelh) [inline software/fastqc/meta.yml:36]: Need to add `single_end` here too.

File: software/fastqc/meta.yml [hunk 6]
```diff
@@ -1,33 +1,48 @@
...
+  - reads:
+      type: file
+      description: Input FastQ file, or pair of files
```
Comment (drpatelh) [inline software/fastqc/meta.yml:39]:
```suggestion
      description: List of input FastQ files of size 1 and 2 for single-end and paired-end data, respectively.
```

File: software/fastqc/meta.yml [hunk 7]
```diff
@@ -1,33 +1,48 @@
...
+  - report:
+      type: file
+      description: FastQC report
+      pattern: "*_fastqc.{zip,html}"
 authors:
-    - "@ewels"
-    - "@FelixKrueger"
+  - "@grst"
+  - "@ewels"
+  - "@FelixKrueger"
```
Comment (drpatelh) [inline software/fastqc/meta.yml:48]:
```suggestion
  - "@grst"
  - "@drpatelh"
  - "@ewels"
  - "@FelixKrueger"
```

File: software/fastqc/test/main.nf [hunk 8]
```diff
@@ -1,21 +1,31 @@
...
-ch_read_files = Channel 
-    .fromFilePairs('../../../test-datasets/test*{1,2}.fastq.gz',size:-1)
-    // .view()  // to check whether the input channel works
+/**
+ * Test if FASTQC runs with single-end data
+ */
+workflow test_single_end {
+    input_files = Channel.fromPath("data/test_single_end.fastq.gz")
+                    .map {f -> [f.baseName, f]}
+    input_files.view()
```
Comment (drpatelh) [inline software/fastqc/test/main.nf:16]:
```suggestion
```

**Pull request merged**